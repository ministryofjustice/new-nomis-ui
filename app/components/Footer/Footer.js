import React, { Fragment } from 'react'
import PropTypes from 'prop-types'
import {
  StyledFooter,
  StyledContainer,
  StyledMeta,
  StyledMetaItem,
  StyledMetaCustom,
  StyledInlineList,
  StyledLicenseDescription,
  StyledLicenseLogo,
  StyledCopyrightLogo,
  StyledHiddenHeader,
  StyledNavigation,
  StyledSection,
  StyledSectionBreak,
  StyledFooterList,
  StyledSectionHeading,
} from './Footer.styles'

import FooterLink from './elements/FooterLink'

const hyphenateString = str => str.replace(/ +/g, '-').toLowerCase()

// https://github.com/alphagov/govuk-frontend/blob/master/src/components/footer/template.njk
const Footer = ({ navigation, meta, children }) => (
  <StyledFooter role="contentinfo">
    <StyledContainer>
      {navigation && navigation.length > 0 && (
        <Fragment>
          <StyledNavigation>
            {navigation.map(section => (
              <StyledSection key={hyphenateString(section.title)}>
                <StyledSectionHeading level={2} size="MEDIUM">
                  {section.title}
                </StyledSectionHeading>
                {section.items && section.items.length > 0 && (
                  <StyledFooterList columns={section.columns}>
                    {section.items.map(item => (
                      <li key={hyphenateString(item.text)}>
                        <FooterLink href={item.href}>{item.text}</FooterLink>
                      </li>
                    ))}
                  </StyledFooterList>
                )}
              </StyledSection>
            ))}
          </StyledNavigation>
          <StyledSectionBreak />
        </Fragment>
      )}

      <StyledMeta>
        <StyledMetaItem grow>
          {meta && meta.items && meta.items.length > 0 && (
            <Fragment>
              <StyledHiddenHeader level={2}>Support links</StyledHiddenHeader>
              <StyledInlineList>
                {meta.items.map(item => (
                  <li key={hyphenateString(item.text)}>
                    <FooterLink href={item.href} {...item}>
                      {item.text}
                    </FooterLink>
                  </li>
                ))}
              </StyledInlineList>
            </Fragment>
          )}
          {children && <StyledMetaCustom>{children}</StyledMetaCustom>}
          <StyledLicenseLogo
            role="presentation"
            focusable="false"
            xmlns="http://www.w3.org/2000/svg"
            viewBox="0 0 483.2 195.7"
            height="17"
            width="41"
          >
            <path
              fill="currentColor"
              d="M421.5 142.8V.1l-50.7 32.3v161.1h112.4v-50.7zm-122.3-9.6A47.12 47.12 0 0 1 221 97.8c0-26 21.1-47.1 47.1-47.1 16.7 0 31.4 8.7 39.7 21.8l42.7-27.2A97.63 97.63 0 0 0 268.1 0c-36.5 0-68.3 20.1-85.1 49.7A98 98 0 0 0 97.8 0C43.9 0 0 43.9 0 97.8s43.9 97.8 97.8 97.8c36.5 0 68.3-20.1 85.1-49.7a97.76 97.76 0 0 0 149.6 25.4l19.4 22.2h3v-87.8h-80l24.3 27.5zM97.8 145c-26 0-47.1-21.1-47.1-47.1s21.1-47.1 47.1-47.1 47.2 21 47.2 47S123.8 145 97.8 145"
            />
          </StyledLicenseLogo>
          <StyledLicenseDescription>
            All content is available under the{' '}
            <FooterLink href="https://www.nationalarchives.gov.uk/doc/open-government-licence/version/3/" rel="license">
              Open Government Licence v3.0
            </FooterLink>
            , except where otherwise stated
          </StyledLicenseDescription>
        </StyledMetaItem>
        <StyledMetaItem>
          <StyledCopyrightLogo href="https://www.nationalarchives.gov.uk/information-management/re-using-public-sector-information/uk-government-licensing-framework/crown-copyright/">
            Â© Crown copyright
          </StyledCopyrightLogo>
        </StyledMetaItem>
      </StyledMeta>
    </StyledContainer>
  </StyledFooter>
)

Footer.propTypes = {
  navigation: PropTypes.arrayOf(
    PropTypes.shape({
      title: PropTypes.string,
      columns: PropTypes.number,
      items: PropTypes.arrayOf(
        PropTypes.shape({
          href: PropTypes.string,
          text: PropTypes.string.isRequired,
        })
      ),
    })
  ),
  meta: PropTypes.shape({
    items: PropTypes.arrayOf(
      PropTypes.shape({
        href: PropTypes.string,
        text: PropTypes.string.isRequired,
      })
    ),
  }),
  children: PropTypes.node,
}

Footer.defaultProps = {
  navigation: [],
  meta: null,
  children: null,
}

Footer.Link = FooterLink

export default Footer
