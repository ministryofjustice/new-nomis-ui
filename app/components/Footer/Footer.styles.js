import styled, { css } from 'react-emotion'
import { SPACING, LINE_HEIGHT, FONT_SIZE, GUTTER_HALF, NTA_LIGHT, BREAKPOINTS, GUTTER } from '@govuk-react/constants'
import { GREY_2, FOOTER_BACKGROUND, FOOTER_TEXT, FOOTER_BORDER_TOP } from 'govuk-colours'
import Header from '@govuk-react/header'
import { RESPONSIVE_4, RESPONSIVE_5, RESPONSIVE_7, RESPONSIVE_8 } from '@govuk-react/constants/lib/spacing'
import crestLogo from './images/govuk-crest.png'
import crestLogo2x from './images/govuk-crest-2x.png'
import { footerLinkStyle } from './elements/FooterLink'

// https://github.com/alphagov/govuk-frontend/blob/master/src/components/footer/_footer.scss

const crestImageWidth2x = 250
const crestImageHeight2x = 204
const crestImageWidth = crestImageWidth2x / 2
const crestImageHeight = crestImageHeight2x / 2

export const StyledFooter = styled('footer')`
  display: flex;
  justify-content: center;
  padding-top: ${RESPONSIVE_7.mobile}px;
  padding-bottom: ${RESPONSIVE_5.mobile}px;
  font-family: ${NTA_LIGHT};
  font-size: ${FONT_SIZE.SIZE_14};
  line-height: ${LINE_HEIGHT.SIZE_14};
  @media screen and (min-width: ${BREAKPOINTS.TABLET}) {
    padding-top: ${RESPONSIVE_7.tablet}px;
    padding-bottom: 25px;
    font-size: ${FONT_SIZE.SIZE_16};
    line-height: ${LINE_HEIGHT.SIZE_16};
  }

  border-top: 1px solid ${FOOTER_BORDER_TOP};
  color: ${FOOTER_TEXT};
  background: ${FOOTER_BACKGROUND};
`

export const StyledContainer = styled('div')`
  width: 100%;
  max-width: 1170px; /* Bespoke to NN - Would ordinarily be SITE_WIDTH_PX */
  margin: 0 ${SPACING.SCALE_3};

  @media screen and (min-width: ${BREAKPOINTS.TABLET}) {
    margin: 0 ${SPACING.SCALE_5};
  }
`

export const StyledSectionBreak = styled('hr')`
  margin: 0;
  margin-bottom: ${RESPONSIVE_8.mobile}px;
  border: 0;
  border-bottom: 1px solid ${GREY_2};

  @media screen and (min-width: ${BREAKPOINTS.DESKTOP}) {
    margin-bottom: ${RESPONSIVE_8.tablet}px;
  }
`

export const StyledMeta = styled('div')`
  display: flex;
  margin-right: -${GUTTER_HALF};
  margin-left: -${GUTTER_HALF};
  flex-wrap: wrap;
  align-items: flex-end;
  justify-content: center;
`

const flexBasisPartial = props =>
  props.grow &&
  css`
    @media screen and (max-width: ${BREAKPOINTS.DESKTOP}) {
      flex-basis: 320px;
    }
  `
export const StyledMetaItem = styled('div')`
  margin-right: ${GUTTER_HALF};
  margin-bottom: 25px;
  margin-left: ${GUTTER_HALF};
  flex: ${props => (props.grow ? 1 : undefined)};
  ${flexBasisPartial}
`

export const StyledMetaCustom = styled('div')`
  margin-bottom: ${SPACING.SCALE_4};
`

// .govuk-visually-hidden
export const StyledHiddenHeader = styled(Header)`
  position: absolute !important;
  width: 1px !important;
  height: 1px !important;
  margin: -1px !important;
  overflow: hidden !important;
  clip: rect(0 0 0 0) !important;
  clip-path: inset(50%) !important;
  white-space: nowrap !important;
`

export const StyledInlineList = styled('div')`
  margin-top: 0;
  margin-bottom: ${SPACING.SCALE_3};
  padding: 0;

  li {
    display: inline-block;
    margin-right: ${SPACING.SCALE_3};
    margin-bottom: ${SPACING.SCALE_1};
  }
`

export const StyledSectionHeading = styled(Header)`
  border-bottom: 1px solid ${GREY_2};
  padding-bottom: ${SPACING.SCALE_2};
  margin-bottom: ${RESPONSIVE_7.mobile}px;

  @media screen and (min-width: ${BREAKPOINTS.DESKTOP}) {
    padding-bottom: ${SPACING.SCALE_4};
    margin-bottom: ${RESPONSIVE_7.tablet}px;
  }
`

export const StyledNavigation = styled('div')`
  display: flex;
  margin-right: -${GUTTER_HALF};
  margin-left: -${GUTTER_HALF};
  flex-wrap: wrap;
`

export const StyledSection = styled('div')`
  display: inline-block;
  margin-right: ${GUTTER_HALF};
  margin-bottom: ${GUTTER};
  margin-left: ${GUTTER_HALF};
  vertical-align: top;

  /* Ensure columns take up equal width (typically one-half:one-half) */
  flex-grow: 1;
  flex-shrink: 1;

  /* Make sure columns do not drop below 200px in width
      Will typically result in wrapping, and end up in a single column on smaller screens. */
  flex-basis: 200px;

  @media screen and (min-width: ${BREAKPOINTS.DESKTOP}) {
    flex-basis: auto;

    &:first-child {
      flex-grow: 2;
    }
  }
`

export const StyledFooterList = styled('ul')`
  margin: 0;
  padding: 0;
  list-style: none;
  column-gap: ${GUTTER};

  @media screen and (min-width: ${BREAKPOINTS.DESKTOP}) {
    /* 2 or 3 columns only */
    column-count: ${props => (props.columns ? props.columns : undefined)};
  }

  li {
    margin-bottom: ${RESPONSIVE_4.mobile}px;

    @media screen and (min-width: ${BREAKPOINTS.TABLET}) {
      margin-bottom: ${RESPONSIVE_4.tablet}px;
    }

    &:last-child {
      margin-bottom: 0;
    }
  }
`

export const StyledLicenseLogo = styled('svg')`
  display: inline-block;
  margin-right: ${SPACING.SCALE_2};
  vertical-align: top;

  @media screen and (max-width: ${BREAKPOINTS.DESKTOP}) {
    margin-bottom: ${SPACING.SCALE_3};
  }
`

export const StyledLicenseDescription = styled('span')`
  display: inline-block;
`

export const StyledCopyrightLogo = styled('a')`
  ${footerLinkStyle}
  display: inline-block;
  min-width: ${crestImageWidth}px;
  padding-top: ${crestImageHeight + 10}px;
  background-image: url(${crestLogo});
  @media only screen and (-webkit-min-device-pixel-ratio: 2),
    not all,
    not all,
    only screen and (min-resolution: 192dpi),
    only screen and (min-resolution: 2dppx) {
    background-image: url(${crestLogo2x});
  }
  background-repeat: no-repeat;
  background-position: 50% 0%;
  background-size: ${crestImageWidth}px ${crestImageHeight}px;
  text-align: center;
  text-decoration: none;
  white-space: nowrap;
`
