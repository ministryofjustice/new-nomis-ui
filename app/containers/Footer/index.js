import React from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import { createSelector, createStructuredSelector } from 'reselect'
import { showTerms } from 'globalReducers/app'
import { linkOnClick } from '../../helpers'
import './footer.scss'

const Footer = ({ showTermsAndConditions, mailTo }) => (
  <footer className="FooterContainer">
    <div className="footer-content">
      <div className="FooterLinksContainer">
        <div className="FooterLink">
          <a {...linkOnClick(showTermsAndConditions)}>Terms and conditions</a>
        </div>
        <div className="FooterLink">
          <a href={`mailto:${mailTo}`}>
            Contact us:&nbsp;&nbsp;
            {mailTo}
          </a>
        </div>
      </div>
      <div className="FooterSignature" />
    </div>
  </footer>
)

Footer.propTypes = {
  showTermsAndConditions: PropTypes.func.isRequired,
  mailTo: PropTypes.string,
}

Footer.defaultProps = {
  mailTo: '',
}

const selectMailTo = () => createSelector(state => state.get('app'), appState => appState.get('mailTo'))

const mapStateToProps = createStructuredSelector({
  mailTo: selectMailTo(),
})

const mapDispatchToProps = dispatch => ({
  showTermsAndConditions: () => dispatch(showTerms()),
})

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Footer)
