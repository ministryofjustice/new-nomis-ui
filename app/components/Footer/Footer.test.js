import React from 'react'
import renderer from 'react-test-renderer'
import { MemoryRouter } from 'react-router-dom'
import Footer from './Footer'

describe('<Footer />', () => {
  it('should match the default snapshot', () => {
    const wrapper = renderer.create(<Footer />).toJSON()

    expect(wrapper).toMatchSnapshot()
  })

  it('should match the with children snapshot', () => {
    const wrapper = renderer
      .create(
        <Footer>
          Built by the <a href="/gds">Government Digital Service</a>
        </Footer>
      )
      .toJSON()

    expect(wrapper).toMatchSnapshot()
  })

  it('should match the with meta snapshot', () => {
    const meta = {
      items: [
        { href: '/normal-link', text: 'Normal Link' },
        { to: '/router-link', text: 'Router Link' },
        { clickHandler: jest.fn(), text: 'onClick Link' },
      ],
    }
    const wrapper = renderer
      .create(
        <MemoryRouter>
          <Footer meta={meta} />
        </MemoryRouter>
      )
      .toJSON()

    expect(wrapper).toMatchSnapshot()
  })

  it('should match the with navigation snapshot', () => {
    const navigation = [
      {
        title: 'Services and information',
        columns: 2,
        items: [
          { href: '/benefits', text: 'Benefits' },
          { href: '/births-deaths', text: 'Births, deaths, marriages and care' },
          { href: '/business-self-employed', text: 'Business and self-employed' },
          { href: '/childcare-parenting', text: 'Childcare and parenting' },
          { href: '/citizenship', text: 'Citizenship and living in the UK' },
          { href: '/crime-justice', text: 'Crime, justice and the law' },
          { href: '/disabled-people', text: 'Disabled people' },
          { href: '/driving-transport', text: 'Driving and transport' },
          { href: '/education-learning', text: 'Education and learning' },
          { href: '/employing-people', text: 'Employing people' },
          { href: '/environment-countryside', text: 'Environment and countryside' },
          { href: '/housing-local', text: 'Housing and local services' },
          { href: '/money-tax', text: 'Money and tax' },
          { href: '/passports-abroad', text: 'Passports, travel and living abroad' },
          { href: '/visas-immigration', text: 'Visas and immigration' },
          { href: '/working-pensions', text: 'Working, jobs and pensions' },
        ],
      },
      {
        title: 'Departments and policy',
        items: [
          { href: '/how-gov-works', text: 'How government works' },
          { href: '/departments', text: 'Departments' },
          { href: '/worldwide', text: 'Worldwide' },
          { href: '/policies', text: 'Policies' },
          { href: '/publications', text: 'Publications' },
          { href: '/announcements', text: 'Announcements' },
        ],
      },
    ]
    const wrapper = renderer.create(<Footer navigation={navigation} />).toJSON()

    expect(wrapper).toMatchSnapshot()
  })
})
