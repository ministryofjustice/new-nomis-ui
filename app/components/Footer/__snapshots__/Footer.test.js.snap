// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> should match the default snapshot 1`] = `
<footer
  className="css-1ynevfo"
  role="contentinfo"
>
  <div
    className="css-1fmqkun"
  >
    <div
      className="css-dzhhuw"
    >
      <div
        className="css-1g6dja6"
      >
        <svg
          className="css-11k3glg"
          focusable="false"
          height="17"
          role="presentation"
          viewBox="0 0 483.2 195.7"
          width="41"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M421.5 142.8V.1l-50.7 32.3v161.1h112.4v-50.7zm-122.3-9.6A47.12 47.12 0 0 1 221 97.8c0-26 21.1-47.1 47.1-47.1 16.7 0 31.4 8.7 39.7 21.8l42.7-27.2A97.63 97.63 0 0 0 268.1 0c-36.5 0-68.3 20.1-85.1 49.7A98 98 0 0 0 97.8 0C43.9 0 0 43.9 0 97.8s43.9 97.8 97.8 97.8c36.5 0 68.3-20.1 85.1-49.7a97.76 97.76 0 0 0 149.6 25.4l19.4 22.2h3v-87.8h-80l24.3 27.5zM97.8 145c-26 0-47.1-21.1-47.1-47.1s21.1-47.1 47.1-47.1 47.2 21 47.2 47S123.8 145 97.8 145"
            fill="currentColor"
          />
        </svg>
        <span
          className="css-7kp13n"
        >
          All content is available under the
           
          <a
            className="css-15shojx"
            href="https://www.nationalarchives.gov.uk/doc/open-government-licence/version/3/"
            to={undefined}
          >
            Open Government Licence v3.0
          </a>
          , except where otherwise stated
        </span>
      </div>
      <div
        className="css-3cr27l"
      >
        <a
          className="css-17cy2up"
          href="https://www.nationalarchives.gov.uk/information-management/re-using-public-sector-information/uk-government-licensing-framework/crown-copyright/"
        >
          © Crown copyright
        </a>
      </div>
    </div>
  </div>
</footer>
`;

exports[`<Footer /> should match the with children snapshot 1`] = `
<footer
  className="css-1ynevfo"
  role="contentinfo"
>
  <div
    className="css-1fmqkun"
  >
    <div
      className="css-dzhhuw"
    >
      <div
        className="css-1g6dja6"
      >
        <div
          className="css-op3zxg"
        >
          Built by the 
          <a
            href="/gds"
          >
            Government Digital Service
          </a>
        </div>
        <svg
          className="css-11k3glg"
          focusable="false"
          height="17"
          role="presentation"
          viewBox="0 0 483.2 195.7"
          width="41"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M421.5 142.8V.1l-50.7 32.3v161.1h112.4v-50.7zm-122.3-9.6A47.12 47.12 0 0 1 221 97.8c0-26 21.1-47.1 47.1-47.1 16.7 0 31.4 8.7 39.7 21.8l42.7-27.2A97.63 97.63 0 0 0 268.1 0c-36.5 0-68.3 20.1-85.1 49.7A98 98 0 0 0 97.8 0C43.9 0 0 43.9 0 97.8s43.9 97.8 97.8 97.8c36.5 0 68.3-20.1 85.1-49.7a97.76 97.76 0 0 0 149.6 25.4l19.4 22.2h3v-87.8h-80l24.3 27.5zM97.8 145c-26 0-47.1-21.1-47.1-47.1s21.1-47.1 47.1-47.1 47.2 21 47.2 47S123.8 145 97.8 145"
            fill="currentColor"
          />
        </svg>
        <span
          className="css-7kp13n"
        >
          All content is available under the
           
          <a
            className="css-15shojx"
            href="https://www.nationalarchives.gov.uk/doc/open-government-licence/version/3/"
            to={undefined}
          >
            Open Government Licence v3.0
          </a>
          , except where otherwise stated
        </span>
      </div>
      <div
        className="css-3cr27l"
      >
        <a
          className="css-17cy2up"
          href="https://www.nationalarchives.gov.uk/information-management/re-using-public-sector-information/uk-government-licensing-framework/crown-copyright/"
        >
          © Crown copyright
        </a>
      </div>
    </div>
  </div>
</footer>
`;

exports[`<Footer /> should match the with meta snapshot 1`] = `
<footer
  className="css-1ynevfo"
  role="contentinfo"
>
  <div
    className="css-1fmqkun"
  >
    <div
      className="css-dzhhuw"
    >
      <div
        className="css-1g6dja6"
      >
        <h2
          className="css-a97mk7-StyledHeader-StyledHoc e4u1u2g0"
          size={undefined}
        >
          Support links
        </h2>
        <div
          className="css-z5k0ly"
        >
          <li>
            <a
              className="css-15shojx"
              href="/normal-link"
              to={undefined}
            >
              Normal Link
            </a>
          </li>
          <li>
            <a
              className="css-15shojx"
              href="/router-link"
              onClick={[Function]}
            >
              Router Link
            </a>
          </li>
          <li>
            <a
              className="css-15shojx"
              href={undefined}
              onClick={[Function]}
              onKeyDown={[Function]}
              role="link"
              tabIndex={0}
              to={undefined}
            >
              onClick Link
            </a>
          </li>
        </div>
        <svg
          className="css-11k3glg"
          focusable="false"
          height="17"
          role="presentation"
          viewBox="0 0 483.2 195.7"
          width="41"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M421.5 142.8V.1l-50.7 32.3v161.1h112.4v-50.7zm-122.3-9.6A47.12 47.12 0 0 1 221 97.8c0-26 21.1-47.1 47.1-47.1 16.7 0 31.4 8.7 39.7 21.8l42.7-27.2A97.63 97.63 0 0 0 268.1 0c-36.5 0-68.3 20.1-85.1 49.7A98 98 0 0 0 97.8 0C43.9 0 0 43.9 0 97.8s43.9 97.8 97.8 97.8c36.5 0 68.3-20.1 85.1-49.7a97.76 97.76 0 0 0 149.6 25.4l19.4 22.2h3v-87.8h-80l24.3 27.5zM97.8 145c-26 0-47.1-21.1-47.1-47.1s21.1-47.1 47.1-47.1 47.2 21 47.2 47S123.8 145 97.8 145"
            fill="currentColor"
          />
        </svg>
        <span
          className="css-7kp13n"
        >
          All content is available under the
           
          <a
            className="css-15shojx"
            href="https://www.nationalarchives.gov.uk/doc/open-government-licence/version/3/"
            to={undefined}
          >
            Open Government Licence v3.0
          </a>
          , except where otherwise stated
        </span>
      </div>
      <div
        className="css-3cr27l"
      >
        <a
          className="css-17cy2up"
          href="https://www.nationalarchives.gov.uk/information-management/re-using-public-sector-information/uk-government-licensing-framework/crown-copyright/"
        >
          © Crown copyright
        </a>
      </div>
    </div>
  </div>
</footer>
`;

exports[`<Footer /> should match the with navigation snapshot 1`] = `
<footer
  className="css-1ynevfo"
  role="contentinfo"
>
  <div
    className="css-1fmqkun"
  >
    <div
      className="css-thb5ct"
    >
      <div
        className="css-cvoisk"
      >
        <h2
          className="css-1yris73-StyledHeader-StyledHoc e4u1u2g0"
          size="MEDIUM"
        >
          Services and information
        </h2>
        <ul
          className="css-1wreg1n"
        >
          <li>
            <a
              className="css-15shojx"
              href="/benefits"
              to={undefined}
            >
              Benefits
            </a>
          </li>
          <li>
            <a
              className="css-15shojx"
              href="/births-deaths"
              to={undefined}
            >
              Births, deaths, marriages and care
            </a>
          </li>
          <li>
            <a
              className="css-15shojx"
              href="/business-self-employed"
              to={undefined}
            >
              Business and self-employed
            </a>
          </li>
          <li>
            <a
              className="css-15shojx"
              href="/childcare-parenting"
              to={undefined}
            >
              Childcare and parenting
            </a>
          </li>
          <li>
            <a
              className="css-15shojx"
              href="/citizenship"
              to={undefined}
            >
              Citizenship and living in the UK
            </a>
          </li>
          <li>
            <a
              className="css-15shojx"
              href="/crime-justice"
              to={undefined}
            >
              Crime, justice and the law
            </a>
          </li>
          <li>
            <a
              className="css-15shojx"
              href="/disabled-people"
              to={undefined}
            >
              Disabled people
            </a>
          </li>
          <li>
            <a
              className="css-15shojx"
              href="/driving-transport"
              to={undefined}
            >
              Driving and transport
            </a>
          </li>
          <li>
            <a
              className="css-15shojx"
              href="/education-learning"
              to={undefined}
            >
              Education and learning
            </a>
          </li>
          <li>
            <a
              className="css-15shojx"
              href="/employing-people"
              to={undefined}
            >
              Employing people
            </a>
          </li>
          <li>
            <a
              className="css-15shojx"
              href="/environment-countryside"
              to={undefined}
            >
              Environment and countryside
            </a>
          </li>
          <li>
            <a
              className="css-15shojx"
              href="/housing-local"
              to={undefined}
            >
              Housing and local services
            </a>
          </li>
          <li>
            <a
              className="css-15shojx"
              href="/money-tax"
              to={undefined}
            >
              Money and tax
            </a>
          </li>
          <li>
            <a
              className="css-15shojx"
              href="/passports-abroad"
              to={undefined}
            >
              Passports, travel and living abroad
            </a>
          </li>
          <li>
            <a
              className="css-15shojx"
              href="/visas-immigration"
              to={undefined}
            >
              Visas and immigration
            </a>
          </li>
          <li>
            <a
              className="css-15shojx"
              href="/working-pensions"
              to={undefined}
            >
              Working, jobs and pensions
            </a>
          </li>
        </ul>
      </div>
      <div
        className="css-cvoisk"
      >
        <h2
          className="css-1yris73-StyledHeader-StyledHoc e4u1u2g0"
          size="MEDIUM"
        >
          Departments and policy
        </h2>
        <ul
          className="css-ul2glb"
        >
          <li>
            <a
              className="css-15shojx"
              href="/how-gov-works"
              to={undefined}
            >
              How government works
            </a>
          </li>
          <li>
            <a
              className="css-15shojx"
              href="/departments"
              to={undefined}
            >
              Departments
            </a>
          </li>
          <li>
            <a
              className="css-15shojx"
              href="/worldwide"
              to={undefined}
            >
              Worldwide
            </a>
          </li>
          <li>
            <a
              className="css-15shojx"
              href="/policies"
              to={undefined}
            >
              Policies
            </a>
          </li>
          <li>
            <a
              className="css-15shojx"
              href="/publications"
              to={undefined}
            >
              Publications
            </a>
          </li>
          <li>
            <a
              className="css-15shojx"
              href="/announcements"
              to={undefined}
            >
              Announcements
            </a>
          </li>
        </ul>
      </div>
    </div>
    <hr
      className="css-uf3sj0"
    />
    <div
      className="css-dzhhuw"
    >
      <div
        className="css-1g6dja6"
      >
        <svg
          className="css-11k3glg"
          focusable="false"
          height="17"
          role="presentation"
          viewBox="0 0 483.2 195.7"
          width="41"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M421.5 142.8V.1l-50.7 32.3v161.1h112.4v-50.7zm-122.3-9.6A47.12 47.12 0 0 1 221 97.8c0-26 21.1-47.1 47.1-47.1 16.7 0 31.4 8.7 39.7 21.8l42.7-27.2A97.63 97.63 0 0 0 268.1 0c-36.5 0-68.3 20.1-85.1 49.7A98 98 0 0 0 97.8 0C43.9 0 0 43.9 0 97.8s43.9 97.8 97.8 97.8c36.5 0 68.3-20.1 85.1-49.7a97.76 97.76 0 0 0 149.6 25.4l19.4 22.2h3v-87.8h-80l24.3 27.5zM97.8 145c-26 0-47.1-21.1-47.1-47.1s21.1-47.1 47.1-47.1 47.2 21 47.2 47S123.8 145 97.8 145"
            fill="currentColor"
          />
        </svg>
        <span
          className="css-7kp13n"
        >
          All content is available under the
           
          <a
            className="css-15shojx"
            href="https://www.nationalarchives.gov.uk/doc/open-government-licence/version/3/"
            to={undefined}
          >
            Open Government Licence v3.0
          </a>
          , except where otherwise stated
        </span>
      </div>
      <div
        className="css-3cr27l"
      >
        <a
          className="css-17cy2up"
          href="https://www.nationalarchives.gov.uk/information-management/re-using-public-sector-information/uk-government-licensing-framework/crown-copyright/"
        >
          © Crown copyright
        </a>
      </div>
    </div>
  </div>
</footer>
`;
